diff -urN b/boost/stacktrace/detail/frame_unwind.ipp a/boost/stacktrace/detail/frame_unwind.ipp
--- b/boost/stacktrace/detail/frame_unwind.ipp	2017-12-14 05:26:48.000000000 +0530
+++ a/boost/stacktrace/detail/frame_unwind.ipp	2018-12-27 06:30:46.215185973 +0530
@@ -84,7 +84,7 @@
 std::string frame::name() const {
 #if !defined(BOOST_WINDOWS) && !defined(__CYGWIN__)
     ::Dl_info dli;
-    const bool dl_ok = !!::dladdr(addr_, &dli);
+    const bool dl_ok = !!::dladdr((void*)addr_, &dli);
     if (dl_ok && dli.dli_sname) {
         return boost::core::demangle(dli.dli_sname);
     }
diff -urN b/boost/stacktrace/detail/location_from_symbol.hpp a/boost/stacktrace/detail/location_from_symbol.hpp
--- b/boost/stacktrace/detail/location_from_symbol.hpp	2017-12-14 05:26:48.000000000 +0530
+++ a/boost/stacktrace/detail/location_from_symbol.hpp	2018-12-27 06:30:46.215185973 +0530
@@ -28,7 +28,7 @@
     explicit location_from_symbol(const void* addr) BOOST_NOEXCEPT
         : dli_()
     {
-        if (!::dladdr(addr, &dli_)) {
+        if (!::dladdr((void*)addr, &dli_)) {
             dli_.dli_fname = 0;
         }
     }
diff -urN b/libs/filesystem/src/operations.cpp a/libs/filesystem/src/operations.cpp
--- b/libs/filesystem/src/operations.cpp	2017-12-14 05:26:42.000000000 +0530
+++ a/libs/filesystem/src/operations.cpp	2018-12-27 06:31:19.130712586 +0530
@@ -1541,6 +1541,7 @@
 #   if defined(AT_FDCWD) && defined(AT_SYMLINK_NOFOLLOW) \
       && !(defined(__SUNPRO_CC) || defined(__sun) || defined(sun)) \
       && !(defined(linux) || defined(__linux) || defined(__linux__)) \
+      && !(defined(__QNX__) && (_NTO_VERSION <= 700)) \
       && !(defined(__MAC_OS_X_VERSION_MIN_REQUIRED) \
            && __MAC_OS_X_VERSION_MIN_REQUIRED < 101000) \
       && !(defined(__IPHONE_OS_VERSION_MIN_REQUIRED) \
